app/models/world_location.rb

```diff
@@ -13,20 +13,21 @@ def self.reset_cache
 
   def self.all
     cache_fetch("all") do
-      world_locations = Services.worldwide_api.world_locations.with_subsequent_pages.map do |l|
-        new(l) if l.format == "World location" && l.details && l.details.slug.present?
+      world_locations = Services.worldwide_api.world_locations.with_subsequent_pages.map do |response|
+        location = Location.new(response.to_hash)
+        if location.format == "World location" && location.details &&
+            location.slug.present?
+          self.new(location)
+        end
       end
       world_locations.compact
     end
   end
 
   def self.find(location_slug)
     cache_fetch("find_#{location_slug}") do
-      response = Services.worldwide_api.world_location(location_slug)
-      if response
-        data = OpenStruct.new(response.to_hash)
-      end
-      self.new(data) if data
+      response = Services.worldwide_api.world_location(location_slug)&.to_hash
+      self.new(Location.new(response)) if response
```

Comment:
If we get an OpenStruct directly from the adapter, we should be able to rewrite this method with:

```ruby

def self.find(location_slug)
    cache_fetch("find_#{location_slug}") do
      location = Services.worldwide_api.world_location(location_slug)
      self.new(location)
    end
  end
```

We should also be able to remove most of the delegators